lib/rubocop/cop/style/case_condition_repeat.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks that there are no repeated conditions
+      # used in case 'when' expressions
+      #
+      # @example
+      #
+      #    # bad
+      #    case x:
+      #    when 'first'
+      #      do_something
+      #    when 'first'
+      #      do_something_else
+      #
+      # Reports a new offence for each conditional value that is repeated
+      class CaseConditionRepeat < Cop
+        MSG = 'Do not repeat values in case conditionals.'.freeze
+
+        def on_case(case_node)
+          _condition, *whens, _else = *case_node
+          conditions_seen = []
+
+          whens.each do |when_node|
+            conditions = when_expressions(when_node)
+            conditions.each do |cond|
+              if repeated_condition?(conditions_seen, cond)
+                add_offense(case_node, cond.loc.expression, MSG)
+              end
+            end
+            conditions_seen.push(conditions)
+          end
+        end
+
+        private
+
+        def when_expressions(when_node)
```

Comment:
Since we're using `condition` in all other places, I think this should be `#when_conditions`. WDYT? 😀